Source/WTF/wtf/text/StringImpl.cpp

```diff
@@ -332,15 +332,15 @@ Ref<StringImpl> StringImpl::substring(unsigned start, unsigned length)
     return create(span16().subspan(start, length));
 }
 
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
 char32_t StringImpl::characterStartingAt(unsigned i)
 {
     if (is8Bit())
-        return m_data8[i];
-    if (U16_IS_SINGLE(m_data16[i]))
-        return m_data16[i];
-    if (i + 1 < m_length && U16_IS_LEAD(m_data16[i]) && U16_IS_TRAIL(m_data16[i + 1]))
-        return U16_GET_SUPPLEMENTARY(m_data16[i], m_data16[i + 1]);
+        return span8()[i];
+    auto span = span16();
+    if (U16_IS_SINGLE(span[i]))
+        return span[i];
+    if (i + 1 < m_length && U16_IS_LEAD(span[i]) && U16_IS_TRAIL(span[i + 1]))
+        return U16_GET_SUPPLEMENTARY(span[i], span[i + 1]);
     return 0;
```

Comment:
Kinda wish this used `U16_GET`, but that has unwanted code that looks back at a previous code unit.